### PR TITLE
implement royalty struct

### LINKED ISSUE #121: recipient and royalty validity check: both must appear
https://github.com/rmrk-team/rmrk-substrate/blob/27f9e27b784d223b9e4ec05dc9817293c6b8e4d6/pallets/rmrk-core/src/functions.rs#L245-L251
don't currently check if royalty is passed but not recipient.  shouldn't be able to pass one but not the other.
recommend we also change "recipient" to "royalty_recipient" as it has confused me what the difference between "recipient" and "owner" are.  and i wrote the function.

### PR DESCRIPTION
**Pull request by bmacer**
Converts internal Royalty information into a struct, tying amount and recipient together, refactors where needed, adds tests
resolves #121

### CODE REVIEW

File: pallets/rmrk-core/src/functions.rs [hunk 1]
```diff
@@ -270,8 +270,8 @@ where
 		_sender: T::AccountId,
 		owner: T::AccountId,
 		collection_id: CollectionId,
-		recipient: Option<T::AccountId>,
-		royalty: Option<Permill>,
+		royalty_recipient: Option<T::AccountId>,
```
Comment (HashWarlock) [inline pallets/rmrk-core/src/functions.rs:273]: Couldn't we instead do this?
```
royalty: Option<RoyaltyInfo<T::AccountId>>
```
Removes the need for the line 285-296 I believe.

**Review by ilionic — APPROVED** [review]
Two general thoughts related to royalties ( not part of this PR ): 
* Permill might not have enough precision to represent royalties with more decimal points
* Having now `RoyaltyInfo` thinking how big of a hassle would be to have multiple royalty recipients..

**Comment by HashWarlock** [comment]
> * RoyaltyInfo
How would multiple Royalty recipients work? Not sure how that looks like.

**Comment by ilionic** [comment]
> How would multiple Royalty recipients work? Not sure how that looks like.

This is probably not needed complication at this point so we can ignore. But in general during mint could be set primary recipient / amount and secondary. There are few real world use-cases so this questions comes back from time to time. ( ie collaborations between artists, or creator + 'genesis' owner )

**Comment by bmacer** [comment]
Going to merge this, but I'd imagine having a BoundedVec of royalty recipients wouldn't be all that difficult to implement.  For another day though.

**Pull request merged**